lib/jekyll/collection.rb

```diff
@@ -210,6 +210,65 @@ def read_document(full_path)
 
     private
 
+    def sort_docs!
+      if @metadata["order"] && @metadata["order"].is_a?(Array)
+        rearrange_docs!
+      elsif @metadata["sort_by"]
+        sort_docs_by_key!
+      else
+        docs.sort!
+      end
+    end
+
+    def sort_docs_by_key!
+      meta_key = @metadata["sort_by"]
```

Comment:
Use `metadata` attr_reader here instead of instance variable.

You should handle the `nil` case as well (when `meta_key` is nil)